src/capture/capture.cpp

```diff
@@ -185,80 +314,166 @@ void CAPTURE_StartVideoCapture()
 void CAPTURE_StopVideoCapture()
 {
 #if (C_SSHOT)
-	if (capturing_video) {
+	switch (capture.state.video) {
+	case CaptureState::Off:
+		LOG_WARNING("CAPTURE: Not capturing video output");
+		break;
+	case CaptureState::Pending:
+		// It's very hard to hit this branch; handling it for
+		// completeness only
+		LOG_MSG("CAPTURE: Cancelling pending video output capture");
+		capture.state.video = CaptureState::Off;
+		break;
+	case CaptureState::InProgress:
 		capture_video_finalise();
-		capturing_video = false;
+		capture.state.video = CaptureState::Off;
 		LOG_MSG("CAPTURE: Stopped capturing video output");
-	} else {
-		LOG_WARNING("CAPTURE: Not capturing video output");
 	}
 #else
 	LOG_WARNING(NoAviSupportMessage);
 #endif
 }
 
-void CAPTURE_AddFrame([[maybe_unused]] const uint16_t width,
-                      [[maybe_unused]] const uint16_t height,
-                      [[maybe_unused]] const uint8_t bits_per_pixel,
-                      [[maybe_unused]] const uint16_t pitch,
-                      [[maybe_unused]] const uint8_t capture_flags,
-                      [[maybe_unused]] const float frames_per_second,
-                      [[maybe_unused]] const uint8_t* image_data,
-                      [[maybe_unused]] const uint8_t* palette_data)
-{
 #if (C_SSHOT)
-	auto image_height = height;
-	if (capture_flags & CaptureFlagDoubleHeight) {
-		image_height *= 2;
+static void handle_add_frame_image_capture(const RenderedImage& image)
```

Comment:
this function looks like there's elegant logic in it but it's visually extremely laden with `capture_` prefixing everything, making it very tough to get a clear grasp on.  I keep getting bogged down:

![2023-06-05_06-20](https://github.com/dosbox-staging/dosbox-staging/assets/1557255/bea84d8f-83be-4c49-9cde-7f593fbe0301)

The fact this is modifying the global state (with things like `capture.state.image.raw = CaptureState::Off;`), says maybe it should become a member function of whatever `capture.` is?  That would eliminate those `capture.` prefixes here.

Reply:
Yeah I was considering converting it to a class. That would get rid most of these prefixes, like in the other image related classes. I'll just do it 😄

Reply:
Godzilla BMI level increasing. :rocket:

Reply:
That's actually a great suggestion, I started factoring out the image capturing stuff from this file into an ImageCapture class, that got rid 90% of the noise. Then I'm gonna rename the current ImageCapture class to ImageSaver which better describes its function anyway. So yeah, great suggestion—I've been just looking at this for far too long so I kinda got used to it 😅 

I'll create a new commit on top of everything for this (so we can compare how it improved things), then we can squash it later. Suggest to hold off reviewing until this lands, probably later today or tomorrow.

Reply:
I've done my best, I think it's a lot better now.